pyscal/pyscallist.py

```diff
@@ -225,10 +225,13 @@ def dump_family_1(self, filename: Optional[str] = None, slgof: bool = False) ->
             family_1_str = self.SWOF()
             keywords = "SWOF"
         if self.pyscaltype == GasOil:
-            family_1_str = self.SGOF()
-            keywords = "SGOF"
             if slgof:
                 logger.warning("SLGOF not meaningful for GasOil. Ignored")
+                family_1_str = ""
```

Comment:
I have some issues with this method:

`def dump_family_1(self, filename: Optional[str] = None, slgof: bool = False) -> str:`

- What is the return value?
- What does a `None` filename signify here? Like a `--dry-run` type thing? Why?
- Why is the check on `self.pyscaltype` not mutually exclusive? I.e. can `self.pyscaltype` be multiple things at the same time? If yes, then this code is very hard to read, because `family_1_str` can be overwritten at any point.
- Move check `if self.pyscaltype == GasWater:` way up.
- If I do `foo.dump_family_1("my_eclipse_file")` where `foo` has length of `0` it does nothing and returns `""`. But what if I do `foo.dump_family_1("my_eclipse_file")` on a non-`0` list and my `self.pyscaltype == GasOil` and `slgof == True`. Isn't that ambiguous?
- When you have multiple of these: `family_1_str = self.SGOF(); keywords = "SGOF"` isn't there a way programmatically join these without repetition?

This probably concerns both `dump_family_1` and `dump_family_2`.

Reply:
Thanks! I suggest to rewrite `dump_family_*` to `make_family_*` and only return a string and not relate to the filesystem. The only user of the `filename` option is `pyscalcli.py`, which probably should do the filesystem interaction. 

(`dump_*` will be left present with a deprecation warning, there might be api users)

`pyscaltype` can only be one thing pr. `pyscallist` instance, it is a class property.

Reply:
Refactoring done. Old tests will be kept for a while. Remains to write new tests for the new functions in this PR.